systemtest/src/test/java/io/strimzi/systemtest/kafka/QuotasST.java

```diff
@@ -0,0 +1,95 @@
+
+public class QuotasST extends AbstractST {
+    private static final Logger LOGGER = LogManager.getLogger(QuotasST.class);
+
+    public static final String NAMESPACE = "quotas";
+    /**
+     * Test to check Kafka Quotas Plugin for disk space
+     */
+    @ParallelNamespaceTest
+    @Tag(INTERNAL_CLIENTS_USED)
+    void testKafkaQuotasPluginIntegration(ExtensionContext extensionContext) {
+        final String namespaceName = extensionContext.getStore(ExtensionContext.Namespace.GLOBAL).get(Constants.NAMESPACE_KEY).toString();
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String topicName = mapWithTestTopics.get(extensionContext.getDisplayName());
+
+        final String producerName = "quotas-producer";
+        final String consumerName = "quotas-consumer";
+
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, 1)
+            .editSpec()
+                .editKafka()
+                    .addToConfig("client.quota.callback.class", "io.strimzi.kafka.quotas.StaticQuotaCallback")
+                    .addToConfig("client.quota.callback.static.storage.hard", "55000000")
+                    .addToConfig("client.quota.callback.static.storage.soft", "50000000")
+                    .addToConfig("client.quota.callback.static.storage.check-interval", "5")
+                    .withNewPersistentClaimStorage()
+                        .withNewSize("1Gi")
+                    .endPersistentClaimStorage()
+                .endKafka()
+            .endSpec()
+            .build());
+        resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(clusterName, topicName).build());
+
+        // Send more messages than disk can store to see if the integration works
+        KafkaBasicExampleClients basicClients = new KafkaBasicExampleClients.Builder()
+            .withProducerName(producerName)
+            .withConsumerName(consumerName)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+            .withTopicName(topicName)
+            .withMessageCount(100000000)
+            .withDelayMs(0)
+            .withMessage(String.join("", Collections.nCopies(1000, "#")))
+            .build();
+
+        resourceManager.createResource(extensionContext, basicClients.producerStrimzi().build());
+        // Kafka Quotas Plugin should stop producer in around 10-20 seconds with configured throughput
+        assertThrows(WaitException.class, () -> JobUtils.waitForJobFailure(producerName, NAMESPACE, 120_000));
+
+        String kafkaLog = kubeClient(namespaceName).logs(KafkaResources.kafkaPodName(clusterName, 0));
+        String softLimitLog = "disk is beyond soft limit";
+        String hardLimitLog = "disk is full";
+        assertThat("Kafka log doesn't contain '" + softLimitLog + "' log", kafkaLog, CoreMatchers.containsString(softLimitLog));
```

Comment:
This might be a contrived test suggestion, but would it be worth writing a similar test where we only hit the soft limit, but not the hard limit?
I guess the question is who owns the soft/hard limiting capability, this test assumes that soft limiting/hard limiting just works, which is a fine assumption probably.
I guess my question is how difficult would it be to engineer a test that checks the soft limit has been passed, but not the hard limit?

Reply:
With the current capabilities of system tests it might be possible to basically duplicate this test and see that producer finished at some point. I think it might be also part of this test, but it opens a lot of space for flakiness from my POV. We agreed that we will more focus on Kafka Quota Plugin tests during releases and our manual verification where it's much more easy without any proper performance tests.

Reply:
I think we can create an issue for some performance tests which could cover it. WDYT?

Reply:
Sorry didn't see your response till now!
I think that sounds more than reasonable,  and I don't think we want to introduce flaky tests, I think manual checks and a trust that the quota plugin works as intended is more than enough.